### PR TITLE
adding support for SRV records

### PR DESCRIPTION
**Pull request by reuven8e14**

### CODE REVIEW

File: src/rr.h [hunk 1]
```diff
@@ -468,6 +468,35 @@ class RDataNAPTR : public RData {
...
 };
 
+/**
+ * SRV Record Representation
+ */
+class RDataSRV : public RData {
+    public:
+        RDataSRV() { };
+        virtual ~RDataSRV() { };
+        virtual eRDataType getType() const { return RDATA_SRV; };
```
Comment (mnezerka) [inline src/rr.h]: Could you remove `const` please? It hides overloaded virtual function, which has different qualifiers (unqualified vs 'const'):
```
virtual eRDataType getType() = 0;
```

File: src/unittests.cpp [hunk 2]
```diff
@@ -291,6 +291,20 @@ void testNAPTR()
     assert (r.getReplacement() == "_sip._tcp.icscf.brn56.iit.ims");
 }
 
+void testSRV()
+{
+    dns::RDataSRV r;
+    char dasrv[] = "\x00\x14\x00\x00\x14\x95\x04\x61\x6c\x74\x32\x0b\x78\x6d\x70\x70\x2d\x73\x65\x72\x76\x65\x72\x01\x6c\x06\x67\x6f\x6f\x67\x6c\x65\x03\x63\x6f\x6d\x00";
+
+    assert (r.getType(), dns::RDATA_SRV);
```
Comment (mnezerka) [inline src/unittests.cpp]: Not sure if you tried to test it, but assert takes one argument (expression) and verifies the result of the expression. At least on my MacBook, where I tried to verify your PR. Could you change it from `,` to `==`, please? Same on following lines in this test function body.

**Review by mnezerka — CHANGES_REQUESTED** [review]
Thanks. Could you address my findings, please? I'll then merge it.

File: src/rr.h (hunk 1 above)
Comment (reuven8e14) [reply src/rr.h]: yup, sure np.

File: src/unittests.cpp (hunk 2 above)
Comment (reuven8e14) [reply src/unittests.cpp]: correct, my mistake forgot to change from our macros check.

**Review by mnezerka — APPROVED** [review]
Looks good. Thank you very much for your participation!

File: src/rr.cpp [hunk 3]
```diff
@@ -432,6 +432,39 @@ std::string RDataNAPTR::asString()
...
+    mPort = buffer.get16bits();
+
+    mTarget.clear();
+    uint posStart = buffer.getPos();
+    while (buffer.getPos() - posStart < size - 6) {
+        mTarget.append(buffer.getDnsCharacterString());
+        mTarget.append(".");
+    }
+    mTarget.pop_back();
+    mTarget.pop_back();
```
Comment (wxiaoguang) [inline src/rr.cpp:449]: hint: the `target` is a domain-name, should be read by `buffer.getDnsDomainName`.

And `mTarget.pop_back();` might cause segfault when processing corrupted message.

Comment (mnezerka) [reply src/rr.cpp:449]: Can you write a unit test that triggers this?

Comment (wxiaoguang) [reply src/rr.cpp:449]: I didn't use this code, I am using something like `getDnsDomainName` already.

To trigger the case, make a SRV record without domain field, then the `while` won't be executed, then mTarget is empty, then `mTarget.pop_back()` will crash. https://en.cppreference.com/w/cpp/string/basic_string/pop_back : The behavior is undefined if the string is empty.

**Pull request merged**